Create Release Post 3.40 (#5348)
sourcegraph/about: blogposts/2022/release-3.40.md

---
title: 'Sourcegraph 3.40 release'
publishDate: 2022-05-20T10:00-07:00
description: 'Sourcegraph 3.40 introduces...'
tags: [blog, release]
slug: 'release/3.40'
published: false
heroImage: https://storage.googleapis.com/sourcegraph-assets/blog/3.40/sourcegraph-3-40-release.png
socialImage: https://storage.googleapis.com/sourcegraph-assets/blog/3.40/sourcegraph-3-40-release.png
changelogItems:
  - description:
    url:
    category:
---

> Sourcegraph 3.40 is now available! Here are some highlights from this release:

Reviewer: ```suggestion
Sourcegraph 3.40 is now available! Here are some highlights from this release:

# Local navigation for the top 8 languages is now faster and more accurate

We [previously improved Java](https://about.sourcegraph.com/blog/release/3.39/#Local-navigation-for-Java-code-is-now-faster-and-more-accurate), and since then we’ve added support for Go, C#, Python, JavaScript, TypeScript, C/C++, and Ruby.
```

---
title: 'Sourcegraph 3.40 release'
publishDate: 2022-05-20T10:00-07:00
description: 'Sourcegraph 3.40 introduces...'
tags: [blog, release]
slug: 'release/3.40'
published: false
heroImage: https://storage.googleapis.com/sourcegraph-assets/blog/3.40/sourcegraph-3-40-release.png
socialImage: https://storage.googleapis.com/sourcegraph-assets/blog/3.40/sourcegraph-3-40-release.png
changelogItems:
  - description:
    url:
    category:
---

Sourcegraph 3.40 is now available! Here are some highlights from this release:

# Local navigation for the top 8 languages is now faster and more accurate

> We [previously improved Java](https://about.sourcegraph.com/blog/release/3.39/#Local-navigation-for-Java-code-is-now-faster-and-more-accurate), and since then we’ve added support for Go, C#, Python, JavaScript, TypeScript, C/C++, and Ruby.

Author: ```suggestion
We [previously improved navigation speed for Java code](https://about.sourcegraph.com/blog/release/3.39/#Local-navigation-for-Java-code-is-now-faster-and-more-accurate) in Sourcegraph 3.39. We’ve now made the same improvements for Go, C#, Python, JavaScript, TypeScript, C/C++, and Ruby, so you can navigate your code more quickly and precisely.
```

@macraig, I'd like to add a little bit more excitement & impact to this highlight. I made a suggestion — what do you think of this?

## Search your Python dependencies

> Sourcegraph [dependencies search](https://docs.sourcegraph.com/code_search/how-to/dependencies_search) now supports python. You can now natively search your python dependencies to quickly determine if any dependency across your code has a security vulnerability. We currently have support for poetry.lock and pipfile.lock files. Check out [this example search on Sourcegraph Cloud](https://sourcegraph.com/search?q=context:global+repo:deps%28%5Egithub%5C.com/textualize/rich%24%29+&patternType=literal).

Author: ```suggestion
Sourcegraph [dependencies search](https://docs.sourcegraph.com/code_search/how-to/dependencies_search) now supports Python packages. You can now natively search your Python dependencies to quickly determine if any dependency across your code has a security vulnerability. We currently have support for poetry.lock and pipfile.lock files. Check out [this example search on Sourcegraph Cloud](https://sourcegraph.com/search?q=context:global+repo:deps%28%5Egithub%5C.com/textualize/rich%24%29+&patternType=literal).
```

@eugeniaft, can we add any more detail on this bit? I'm also wondering about the phrasing — are users searching for if their dependencies have vulnerabilities, or are they checking if they're dependent on a package with a known vulnerability?
"if any dependency across your code has a security vulnerability"

Reviewer 2: @iskyOS it would be more the latter: checking if they're dependent on a package with a known vulnerability

## Share links to code insights

[img placeholder]

> You can now link to individual code insight charts directly via the context menu on each insight. When you click to share an insight, the product indicates what other groups – global or individual Sourcegrpah organizations – can see the insight. To share an insight chart, it will need to be on at least one global or org-wide dashboard, and the share link will enforce the permissions of the dashboard(s) with the insight.

Reviewer 3: @iskyOS there will be if you want to stamp this PR! https://github.com/sourcegraph/sourcegraph/pull/35773/files Then we can link to that headed section.
